code/game/machinery/PDApainter.dm

```diff
@@ -103,53 +104,59 @@
 
+/obj/machinery/pdapainter/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	if(isnull(held_item))
+		context[SCREENTIP_CONTEXT_LMB] = "Open UI"
+		if(stored_pda)
+			context[SCREENTIP_CONTEXT_RMB] = "Eject PDA"
+		else if(stored_id_card)
+			context[SCREENTIP_CONTEXT_RMB] = "Eject ID"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	if(istype(held_item, /obj/item/modular_computer/pda))
+		context[SCREENTIP_CONTEXT_LMB] = "[stored_pda ? "Swap" : "Insert"] PDA"
+		return CONTEXTUAL_SCREENTIP_SET
+	if(isidcard(held_item))
+		context[SCREENTIP_CONTEXT_LMB] = "[stored_id_card ? "Swap" : "Insert"] ID"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	switch(held_item.tool_behaviour)
+		if(TOOL_WRENCH)
+			context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Una" : "A"]nchor"
+			return CONTEXTUAL_SCREENTIP_SET
+		if(TOOL_WELDER)
+			context[SCREENTIP_CONTEXT_LMB] = "Repair"
+			return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
 /obj/machinery/pdapainter/wrench_act(mob/living/user, obj/item/tool)
 	. = ..()
 	if(default_unfasten_wrench(user, tool))
 		power_change()
 	return ITEM_INTERACT_SUCCESS
 
-/obj/machinery/pdapainter/attackby(obj/item/O, mob/living/user, list/modifiers, list/attack_modifiers)
-	if(machine_stat & BROKEN)
-		if(O.tool_behaviour == TOOL_WELDER && !user.combat_mode)
-			if(!O.tool_start_check(user, amount=1))
-				return
-			user.visible_message(span_notice("[user] is repairing [src]."), \
-							span_notice("You begin repairing [src]..."), \
-							span_hear("You hear welding."))
-			if(O.use_tool(src, user, 40, volume=50))
-				if(!(machine_stat & BROKEN))
-					return
-				to_chat(user, span_notice("You repair [src]."))
-				set_machine_stat(machine_stat & ~BROKEN)
-				atom_integrity = max_integrity
-				update_appearance(UPDATE_ICON)
-			return
-		return ..()
-
-	// Chameleon checks first so they can exit the logic early if they're detected.
-	if(istype(O, /obj/item/card/id/advanced/chameleon))
-		to_chat(user, span_warning("The machine rejects your [O]. This ID card does not appear to be compatible with the PDA Painter."))
-		return
-
-	if(istype(O, /obj/item/modular_computer/pda))
-		insert_pda(O, user)
-		return
+/obj/machinery/pdapainter/welder_act(mob/living/user, obj/item/tool)
+	if(!(machine_stat & BROKEN) && (atom_integrity >= max_integrity))
+		balloon_alert(user, "isn't broken!")
+		return ITEM_INTERACT_BLOCKING
+	if(!tool.tool_start_check(user, amount = 1))
+		balloon_alert(user, "not enough fuel!")
+		return ITEM_INTERACT_BLOCKING
 
-	if(isidcard(O))
-		if(stored_id_card)
-			to_chat(user, span_warning("There is already an ID card inside!"))
-			return
+	if(!tool.use_tool(src, user, 40, volume = 50))
+		return ITEM_INTERACT_BLOCKING
 
-		if(!user.transferItemToLoc(O, src))
-			return
-
-		stored_id_card = O
-		O.add_fingerprint(user)
-		update_appearance(UPDATE_ICON)
-		return
+	set_machine_stat(machine_stat & ~BROKEN)
+	atom_integrity = max_integrity
+	balloon_alert(user, "repaired!")
```

Comment:
```suggestion
	balloon_alert(user, "repaired")
```

Success states shouldn't have any punctuation, silly at times I know but look at https://hackmd.io/@tgstation/SJIle5wYd#balloon_alertballoon_alert_to_viewers for more info